docs/api-reference.md

```diff
@@ -44,6 +44,22 @@ The result is a `RoactTree`, which is an opaque handle that represents a tree of
 
 ---
 
+### Roact.shallow
```

Comment:
When we do the full markdown page on snapshot testing, we should walk through the use cases for when developers should use which methods of shallow wrapper. When should developers assert facts w/ calls to `find` versus only doing a snapshot?